packages/debian/download_packages.py

```diff
@@ -53,6 +54,44 @@ def select_package_version(
     ))
 
 
+def select_package_origin(package: apt.package.Version) -> str:
+    """Select a trusted origin for the package and return its "name"."""
+    for origin in package.origins:
+        if not origin.trusted:
+            continue
+        # Some repo don't have `archive` (Salt, I'm looking at you…).
+        return str(origin.archive) or str(origin.label)
+    raise Exception('no trusted origin found for package {}'\
+                    .format(pathlib.Path(package.filename).name))
+
+
+# We can't use `package.fetch_binary` because it relies on MD5 and some packages
+# only have a SHA256…
+#
+# In the same vein, we are doing checksum validation manually because
+# `apt_pkg.AcquireFile` is failing on some packages (the ones from the external
+# repositories…).
+def fetch_binary(package: apt.package.Version, destdir: pathlib.Path) -> None:
+    """Download the DEB file corresponding to the given package."""
+    destdir.mkdir(exist_ok=True)  # TODO: to be done by the build chain.
+
+    filename = pathlib.Path(package.filename).name
+    destfile = destdir/filename
+    print('Downloading package {}'.format(filename))
```

Comment:
stray print?

Reply:
Nope, on purpose: to have some input during the download (which can take a while on "slow" connection).